docs/architecture/adr-047-extend-upgrade-plan.md

```diff
@@ -3,28 +3,28 @@
 ## Changelog
 
 * Nov, 23, 2021: Initial Draft
+* Feb 15, 2023: Remove `pre_run` and `post_run` fields for simplifying the ADR
```

Comment:
This does not seem needed anymore, is that a correct assumption?

Reply:
This ADR has been in the back of my mind for the last year+. To me, the only thing really worthwhile in it, is the `pre_run` and `post_run` stuff. Cosmovisor currently has support for running a `pre-upgrade` command on the new executable. Support for running an externally defined command before and/or after an upgrade would still be handy, though. An example pre-run is updating config files to a new format (e.g. when all the `-`s change to `_`s). An example post-run is sending a notification somewhere that a node is back up.

That being said, the `post_run` concept might be better as something a node can configure rather than something all nodes on a chain would run. I can't remember all of the use-cases previously thought of for it, though.

As for the `Artifact`s defined in this PR, I actually now think it's something that I _do not want_. We (at Provenance Blockchain) put a URL in the info field that returns the `UpgradeConfig` JSON. We've had a few instances where an upgrade proposal is submitted and passed, but before the upgrade height, we discover a major issue. Because the actual executable download URLs aren't in the proposal, we're able to change them to a version without the issue. Without this ability, it'd often be impossible to prevent the chain from upgrading to the problematic version.

If the `pre_run` and `post_run` parts are being removed from this ADR, I'd suggest getting rid of the ADR altogether.